### PR TITLE
Visual regression test pr 

### PR DESCRIPTION
**Pull request by holderb5**
Resolves #
https://github.com/yahoo/navi/issues/847
Resolves the above for local testing, not for travis-ci

## Description
Integrates 5 basic ember backstop tests into navi, as well as docker connection and capability to run tests reliably in dev mode.
## Proposed Changes

- Adds in capability for ember backstop testing
- Upon cloning of repository running "BACKSTOP=true ember server" will allow developers to navigate to testing and see the backstop tests run, for backstop tests development is required
- docker command: docker run -p 3000:3000 --rm -it --mount type=bind,source="`pwd`",target=/src backstopjs/backstopjs:4.5.1 remote "--moby=true"
- results of tests can be seen at /backstop/backstop_data/html_report/

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/app/backstop.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{
```
Comment (jkusa) [inline packages/app/backstop.json:1]: Is this file used?

Comment (jkusa) [inline packages/app/backstop.json:1]: Is this file used?

File: packages/app/ember-backstop/backstop.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/* eslint-env browser, node */
```
Comment (jkusa) [inline packages/app/ember-backstop/backstop.js:1]: Is this file used?

File: packages/app/package.json [hunk 3]
```diff
@@ -18,42 +18,44 @@
     "lint:js": "eslint .",
     "lint": "npm run lint:hbs && npm run lint:js",
     "start": "ember serve",
+    "pretest": " ember backstop-remote& ",
```
Comment (jkusa) [inline packages/app/package.json:21]: Can we check if the backstop environment var? Also, can we have a script for the docker command?

File: packages/app/package.json [hunk 4]
```diff
@@ -18,42 +18,44 @@
...
     "ember-cli-app-version": "^3.2.0",
-    "ember-cli-babel": "^7.17.2",
-    "ember-cli-code-coverage": "^1.0.0-beta.8",
+    "ember-cli-babel": "^7.21.0",
+    "ember-cli-code-coverage": "^1.0.0-beta.9",
     "ember-cli-dependency-checker": "^3.2.0",
     "ember-cli-dependency-lint": "^1.1.3",
     "ember-cli-eslint": "^5.1.0",
-    "ember-cli-flash": "^1.8.1",
+    "ember-cli-flash": "^1.9.0",
```
Comment (jkusa) [inline packages/app/package.json:47]: Are these version changes intentional?

File: packages/app/backstop.json (hunk 1 above)
Comment (holderb5) [reply packages/app/backstop.json]: Double checked and its not. Removed it now. Same with the backstop js in ember-backstop.

File: packages/app/package.json (hunk 4 above)
Comment (holderb5) [reply packages/app/package.json]: reverted the ones that were not intentional.

File: packages/app/config/dependency-lint.js [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 module.exports = {
   allowedVersions: {
-    'ember-composable-helpers': '3.1.1 || 2.4.0'
+    'ember-composable-helpers': '3.2.0 || 2.4.0',
+    'ember-maybe-in-element': '0.4.0 || 0.2.0'
```
Comment (jkusa) [inline packages/app/config/dependency-lint.js:6]: Do we need both?

Comment (holderb5) [reply packages/app/config/dependency-lint.js:6]: Looking at these, ember dropdown element automatically uses ember-maybe-in-element 0.2.0 as a dependency so it causes an error without the conditional to have either 0.4.0 or 0.2.0

File: packages/app/ember-backstop/backstop.js (hunk 2 above)
Comment (holderb5) [reply packages/app/ember-backstop/backstop.js:1]: As discussed offline, we don't reference it but builds fail without it

**Comment by jkusa** [comment]
Build is not actually passing: https://travis-ci.com/github/yahoo/navi/jobs/415184500

**Comment by jkusa** [comment]
Needs rebase and new screenshots after Denali. Closing for now

**Pull request closed without merging**